networking/gcp/vpc/main.tf

```diff
@@ -0,0 +1,121 @@
+# Configuration in the current provider
+data "google_client_config" "current" {}
```

Comment:
I would prefer specifying at the very list the region explicitly, and maybe the project also.

Reply:
What do you mean by a list?

The objective is that the configuration of the cloud providers are defined at the root module. If we need change the provider for a specific module, we proceed as follows, for example:

```terraform
provider "gcp" {
  region = "europe-west3"
}

provider "gcp" {
  alias  = "ew9"
  region = europe-west9"
}

module "my_module" {
  source    = "./my-module"
  providers = {
    gcp = gcp.ew9
  }
  ...
}
```

Reply:
Oups, I didn't meant "list" but "least"... In GCP, as you can very easily span resources across regions, it is more usual to explicitly specify the region when creating the resource. And this was a question I asked somewhere else: Do we want to bind a region to a VPC, while a VPC is a global resource?
The same can be done with projects also. This is a bit more contrived, but also works.

Reply:
I would add that what you propose is the only way to do it in AWS, but in GCP, all resources can take an optional `project` and `region` parameter, and it makes it super easy to use multiple projects and region inside a single deployment.